### PR TITLE
e2e test speedup

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This eliminates the 40 second wait after each test.

### CODE REVIEW

File: test/e2e/dra/dra.go [hunk 1]
```diff
@@ -249,7 +249,8 @@ func (b *builder) resourceClaimParameters() *corev1.ConfigMap {
 // makePod returns a simple pod with no resource claims.
 // The pod prints its env and waits.
 func (b *builder) pod() *corev1.Pod {
-	pod := e2epod.MakePod(b.f.Namespace.Name, nil, nil, false, "env && sleep 100000")
+	pod := e2epod.MakePod(b.f.Namespace.Name, nil, nil, false, "env")
```
Comment (bart0sh) [inline test/e2e/dra/dra.go]: Can we just reduce sleep to say 10 sec? This way we can catch more races, I think.

Comment (pohly) [reply test/e2e/dra/dra.go]: Because it's more realistic, i.e. pods actually run for a while? Yes, that makes sense to me.

Comment (pohly) [reply test/e2e/dra/dra.go]: Alternatively, we can set `TerminationGracePeriodSeconds` to zero and keep the `sleep 100000`. I've push a commit doing that.

Comment (bart0sh) [reply test/e2e/dra/dra.go]: That would work. However, I don't think it's a realistic expectation that user would reduce grace period to 0. I'd just reduce sleep time as 100000 seconds is too long.

Comment (pohly) [reply test/e2e/dra/dra.go]: It's not expected that users do that, it's just meant to speed up the test once it is done. As it just influences a sleep, I'd expect no real difference. We can also make it short (1s).

So as the goal is to be "realistic", what is better: making sure that multiple pods run in parallel and kill them once they run, or let each of them complete quickly? My concern with the latter is that it makes the tests non-deterministic: sometimes they might run in parallel, sometimes they might not.

I prefer deterministic tests.

Comment (bart0sh) [reply test/e2e/dra/dra.go]: You're right. This way it's more deterministic, but it doesn't cover scenarios when pods complete quickly.
Let's keep it like that and add new tests to cover quick completion use cases later.

**Review by bart0sh — APPROVED** [review]
/lgtm

**Pull request merged**